pages/_app.js

```diff
@@ -2,13 +2,39 @@ import Layout from "@/components/Layout";
 import GlobalStyle from "../styles";
 import { exercises } from "@/lib/exercises.js";
 import { workouts } from "@/lib/workouts";
+import useLocalStorageState from "use-local-storage-state";
+import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
+  const [workoutsList, setWorkoutsList] = useLocalStorageState("workoutsList", {
+    defaultValue: workouts,
+  });
+
+  function handleAddWorkout(newWorkout) {
+    setWorkoutsList([{ id: uid(), ...newWorkout }, ...workoutsList]);
+  }
+
+  function handleEditWorkout(editedWorkout) {
+    setWorkoutsList(
+      workoutsList.map((workout) =>
+        workout.id === editedWorkout.id
+          ? { ...workout, ...editedWorkout }
+          : workout
+      )
+    );
+  }
```

Comment:
As far as I can tell, this works very well in this case because the `exercises` property gets always completely overwritten. It might not work as well if `workout` and `editedWorkout` had more nested properties, as the process is shallow, meaning only the top levels of the objects are merged